### PR TITLE
clean up and align with current template

### PR DESCRIPTION
**Pull request by mariux**
- Update main README to latest format
- Update CODEOWNERS and CONTRIBUTING.md with latest content
- Update build-system to latest
- Import example to test and update test README.md
- Disable docker_build_and_push which leaks AWS credentials in semaphore logs
- Add real example with example code and documentation
- Add index README.md to examples

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,60 +1,91 @@
...
 endif
 
-GREEN  := $(shell tput -Txterm setaf 2)
-YELLOW := $(shell tput -Txterm setaf 3)
-WHITE  := $(shell tput -Txterm setaf 7)
-RESET  := $(shell tput -Txterm sgr0)
+DOCKER_RUN_FLAGS += --rm
+DOCKER_RUN_FLAGS += -v ${PWD}:/app/src
+DOCKER_RUN_FLAGS += -e TF_IN_AUTOMATION
+#DOCKER_RUN_FLAGS += -e USER_UID=$(shell id -u)
```
Comment (soerenmartius) [inline Makefile:34]: shouldn't this be uncommented ?

File: test/ecr_repository_test.go [hunk 2]
```diff
@@ -57,53 +55,53 @@ func TestECRRepository(t *testing.T) {
...
-		})
-
-		if err != nil {
-			t.Fatalf("An error occurred while trying to initiate a new docker clientt %s", err)
-		}
-
-		dockerBuild(t, repo, "v1")
-		dockerPush(t, dockerClient, repo, *authorizationDetails.AuthorizationData[0].AuthorizationToken)
-	})
+	// // Authenticate with ECR and push the image
```
Comment (soerenmartius) [inline test/ecr_repository_test.go:58]: why deactivating the test ?

File: Makefile (hunk 1 above)
Comment (mariux) [reply Makefile:34]: see disabled test that leaks credentials... docker in docker does not work in user mode... tests are kind of broken so reduced the scope for now..

File: test/ecr_repository_test.go (hunk 2 above)
Comment (mariux) [reply test/ecr_repository_test.go:58]: not scope of cleanup.. also thsi test is a full integration test... so removing this makes it a unit test... created a follow up ticket...

Comment (soerenmartius) [reply test/ecr_repository_test.go:58]: Why would you not test the integration of this module then? This feels kind of really wrong to me, sorry. We should rather see how we can get the credentials of the outputs. ( which we can ).

If it's not part of the cleanup, then leave the test and address it in another PR.

Comment (mariux) [reply test/ecr_repository_test.go:58]: the test is not important as a unit tests it still creates the resources. this is a security issue. so temporarily deactivating is the immediate fix. this is a normal flow.. disable the security issue and fix later.

Comment (mariux) [reply test/ecr_repository_test.go:58]: fixing this and making it run is not trivial.

**Comment by mariux** [comment]
removed the security fix

**Pull request merged**